.gitlab/scripts.yml

```
@@ -0,0 +1,203 @@
+    - if [ "${EXPORT_BUILD_DIR}" == "ON" ]; then ninja test_exportbuild; fi
+  dependencies: []
+
+
+.horeka_test_template: &horeka_test
+  stage: test
+  before_script:
+    # Sanity checks
+    - if [ -z "${USE_NAME}" ]; then exit 111; fi
+    - if [ -z "${SLURM_PARTITION}" ]; then exit 222; fi
+    - if [[ ! "${SLURM_GRES}" =~ "^gpu*"  ]]; then export NVIDIA_VISIBLE_DEVICES=void; fi
+  variables:
+    GIT_CLEAN_FLAGS: none
+  script:
+    # We should be reusing the previous image and build directory.
+    # Speedup the tests by limiting the number of OMP threads to CI parallelism.
+    - export OMP_NUM_THREADS=${CI_PARALLELISM}
+    # The test job should have the name name as the image job, only changing
+    # build to test
+    - cd ${CI_PROJECT_DIR}${CI_PROJECT_DIR_SUFFIX}
+    - cd ${CI_JOB_NAME/test/build}
+    - ninja test
+    - ninja test_install
+  cache: []
+
+
+.horeka_benchmark_before_script_template: &horeka_benchmark_before_script
+  # Sanity checks
+  - if [ -z "${USE_NAME}" ]; then exit 111; fi
+  - if [ -z "${SLURM_PARTITION}" ]; then exit 222; fi
+  - if [[ ! "${SLURM_GRES}" =~ "^gpu*"  ]]; then export NVIDIA_VISIBLE_DEVICES=void; fi
+  - if [[ -z "${EXECUTOR}" ]]; then exit 233; fi
+  - if [[ -z "${SYSTEM_NAME}" ]]; then exit 233; fi
+  - if [[ -z "${BENCHMARK}" ]]; then exit 233; fi
+  # Set environment variables
+  - eval $(ssh-agent -s)
+  - echo "${BOT_KEY}" | tr -d '\r' | ssh-add - >/dev/null
+  - mkdir -p ~/.ssh
+  - chmod 700 ~/.ssh
+  - ssh-keyscan -t rsa github.com >>~/.ssh/known_hosts
+  - chmod 644 ~/.ssh/known_hosts
+  - echo -e "Host *\n\tStrictHostKeyChecking no\n\n" > ~/.ssh/config
+  - unset OMP_NUM_THREADS
+  - git config --global user.name "${BOT_USER}"
+  - git config --global user.email "${BOT_EMAIL}"
+
+
+.benchmark_template: &horeka_benchmark
+  before_script: *horeka_benchmark_before_script
+  variables:
+    GIT_CLEAN_FLAGS: none
+  script:
+    # Install rsync
+    - pushd /
+    - git clone https://github.com/WayneD/rsync -b v3.2.4
+    - pushd rsync
+    - ./configure --disable-lz4 --disable-openssl --disable-doc --disable-xxhash --disable-md2man
+    - make -j10 install
+    - popd
+    - popd
+    # Setup ssget
+    - git clone https://github.com/ginkgo-project/ssget
+    - sed -i 's:ARCHIVE_LOCATION="${HOME}/.config/ssget":ARCHIVE_LOCATION="/ssget":' ssget/ssget
+    - export PATH=$PWD/ssget:$PATH
+    # Setup the benchmark
+    - cd ${CI_PROJECT_DIR}${CI_PROJECT_DIR_SUFFIX}
+    - cd ${CI_JOB_NAME}-build
+    - chmod +x benchmark/run_all_benchmarks.sh
+    - cd benchmark
+    - git clone ${BENCHMARK_REPO} data-repo
+    # Use the representative matrix list and launch benchmark
+    - export MATRIX_LIST_FILE=$PWD/data-repo/data/represent.list
+    - ./run_all_benchmarks.sh
+    # Publish the results
+    - rsync -av results/ data-repo/data/
+    - cd data-repo/data/
+    - ./build-list . > list.json
+    - ./agregate < list.json > agregate.json
```

nit: But the executable name might also have to be updated
```suggestion
    - ./aggregate < list.json > aggregate.json
```

---

Good point, but since this requires changing everything, including the `GPE`, I would rather not do this right now.